backend/ringv2/ringv2.go

```diff
@@ -65,6 +72,7 @@ type Event struct {
 	Err error
 }
 
+// Ring is a ring of items that are triggered in a round robin fashion.
```

Comment:
Ring is a circular queue of items that are cooperatively iterated over by one or more subscribers. The subscribers are notified every time an item goes from the front to the back of the queue, with EventTrigger. Iteration proceeds according to an interval, and is triggered by an etcd lease expiring.